backend/src/oe/service/op-block-header.service.ts

```diff
@@ -0,0 +1,137 @@
+
+      return {
+        height,
+        version,
+        chainWork: chainwork,
+        medianTime: mediantime,
+        previousBlockHash: previousblockhash,
+        merkleRoot: merkle_root,
+        timestamp,
+        difficulty,
+        nonce,
+        reward: totalfee + subsidy,
+        currentBlockHash: blockHash
+      };
+    } catch (error) {
+      logger.err(
+        `Error while fetching block header ${blockHeight.value}: ${error}`
+      );
+      throw error;
+    }
+  }
+
+  public async $syncOlderBlockHeader(UUID: string, currentTip?: number): Promise<void> {
+    try {
+      logger.debug('Syncing older block headers');
+
+      let { value: currentSyncedBlockHeight } = await opBlockHeaderRepository.$getLatestBlockHeight(UUID);
+
+
+      // Only using fetching current block tip if not present in argument
+      if (!currentTip) {
+        currentTip = await bitcoinApi.$getBlockHeightTip();
+      }
+      const { value: latestConfirmedBlockHeight } = this.verifyConfirmedBlockHeight(currentTip - config.OP_ENERGY.CONFIRMED_BLOCKS_AMOUNT, { value: currentTip });
+
+
+      logger.debug(
+        `currentSyncedBlockHeight: ${currentSyncedBlockHeight}, latestConfirmedBlockHeight: ${latestConfirmedBlockHeight}`
+      );
+
+      if (currentSyncedBlockHeight >= latestConfirmedBlockHeight) {
+        logger.debug('Already synced block headers.');
+        return;
+      }
+
+      currentSyncedBlockHeight += 1;
+
+      logger.debug(
+        `Syncing block header from #${currentSyncedBlockHeight} to #${latestConfirmedBlockHeight}`
+      );
+
+      while (latestConfirmedBlockHeight >= currentSyncedBlockHeight) {
+        const noOfBlockHeaders = Math.max(
+          1,
+          Math.min(10, latestConfirmedBlockHeight - currentSyncedBlockHeight)
+        );
+
+        const blockHeaders = await Bluebird.map(
+          [...Array(noOfBlockHeaders).keys()].map(
+            (i) => i + currentSyncedBlockHeight
+          ),
+          (height) => this.$getBlockHeaderData({ value: height })
+        );
+
+        await Bluebird.mapSeries(blockHeaders, (blockHeader) =>
+          opBlockHeaderRepository.$saveBlockHeaderInDatabase(UUID, blockHeader as BlockHeader)
+        );
+
+        currentSyncedBlockHeight += noOfBlockHeaders;
+      }
+
+      logger.debug('Synced all the missing block headers');
+    } catch (error) {
+      logger.err(`Something went wrong while syncing block header.` + error);
+    }
+  }
+
+  public async $getBlockHeader(UUID: string, height: number): Promise<BlockHeader | null> {
+    try {
+      const blockHeader = await opBlockHeaderRepository.$getBlock(UUID, height);
+      return blockHeader;
+    } catch (error) {
+      logger.err(`Something went wrong while fetching block header.` + error);
+      throw error;
+    }
+  }
+
+  public async $getBlockHeadersByHeights(UUID: string, bockHeights: number[]): Promise<BlockHeader[]> {
```

Comment:
getBlockHeadersByHeights is used while calculating nbdr I prefer to use the service instead of directly calling the repository in the nbdr service. 

So nbdr service is just consuming another service OpBlockHeaderService